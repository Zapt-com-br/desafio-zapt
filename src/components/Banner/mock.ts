export default {
  title: 'Bem vinda Polyane,',
  contactNumber: '11910898780'
}
