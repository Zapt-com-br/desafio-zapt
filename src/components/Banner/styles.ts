import styled, { css } from 'styled-components'
import { darken } from 'polished'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr auto;
    place-items: center;
    background: ${theme.colors.bgBanner};
    border-radius: ${theme.border.radius};
    position: relative;

    &:before {
      content: '';
      width: 62px;
      height: 62px;
      background: ${theme.colors.secondary};
      position: absolute;
      border-radius: 50% 50% 50% 0% / 50% 50% 50% 0%;
      top: -19px;
      left: -25px;
    }
  `}
`

export const Content = styled.div`
  ${({ theme }) => css``}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.normal};
    color: ${theme.colors.black};
    margin-bottom: ${theme.spacings.xxsmall};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.gray};
    max-width: 58.8rem;
    line-height: ${theme.font.line.medium};
  `}
`

export const ContentAction = styled.div`
  ${({ theme }) => css`
    padding-top: ${theme.spacings.medium};
  `}
`

export const ActionQuestion = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.semibold};
    margin-bottom: ${theme.spacings.xsmall};
    color: ${theme.colors.black};
  `}
`

export const ButtonAction = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: ${theme.border.none};
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;
    transition: all ${theme.transition.default};

    &:hover {
      color: ${darken(0.05, theme.colors.white)};
      background: ${darken(0.05, theme.colors.primary)};
    }
  `}
`

export const ButtonWrapper = styled.div`
  ${({ theme }) => css`
    width: fit-content;
    display: grid;
    grid-template-columns: auto auto;
    gap: ${theme.spacings.small};
  `}
`

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    &:before {
      content: '';
      width: 80px;
      height: 80px;
      background: ${theme.colors.white};
      position: absolute;
      border-radius: 50%;
      top: 16px;
      left: 71px;
    }

    &:after {
      content: '';
      width: 100px;
      height: 100px;
      background: ${theme.colors.primary};
      position: absolute;
      border-radius: 50% 50% 50% 0% / 50% 50% 50% 0%;
      bottom: -30px;
      right: -43px;
    }
  `}
`

export const SemicircleWrapper = styled.div`
  position: absolute;
  right: 38px;
  bottom: 35px;
`

export const BannerImage = styled.img`
  ${({ theme }) => css`
    display: block;
    max-width: 100%;
    max-height: 313px;
    object-fit: cover;
    border-radius: ${theme.border.radius};
  `}
`

export const Footer = styled.footer`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.gray};
    padding: ${theme.spacings.small} 0;
    grid-column: 1/-1;
    background: ${theme.colors.white};
    width: 100%;
  `}
`

export const Link = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    margin-left: 0.4rem;
    text-decoration: underline;
    text-underline-position: under;
    cursor: pointer;
  `}
`
