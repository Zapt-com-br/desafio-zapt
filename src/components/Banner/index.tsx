import * as S from 'components/Banner/styles'
import * as C from 'components'
import Image from 'next/image'
import { useRouter } from 'next/router'
import { ReactNode } from 'react'

export type BannerProps = {
  title: string
  description: string
  question?: string
  img: string
  btn?: { textBtn: string; route: string }[]
  children?: ReactNode
}

const Banner = ({
  title,
  description,
  question,
  img,
  btn,
  children
}: BannerProps) => {
  const router = useRouter()

  return (
    <>
      <S.Container>
        <S.ImgPinkBalloon
          src="/imgs/pink-balloon.svg"
          alt="Balão Rosa"
          width="62"
          height="62"
        />
        <S.Row>
          <S.WrapperInfo>
            <S.Title>{title}</S.Title>
            <S.Description>{description}</S.Description>
            <C.IF condition={!!question}>
              <S.Question>{question}</S.Question>
            </C.IF>

            <C.IF condition={!!btn}>
              <S.BtnGroup>
                {btn?.map(({ textBtn, route }, i) => (
                  <C.Button
                    key={textBtn}
                    color="blue"
                    size="large"
                    text={textBtn}
                    onClick={() => router.push(route)}
                    aria-label={textBtn}
                    className={`btn${i}`}
                  />
                ))}
              </S.BtnGroup>
            </C.IF>
          </S.WrapperInfo>

          <S.WrapperImg>
            <Image
              src={img}
              alt="Banner"
              width="433"
              height="313"
              priority={true}
            />
            <S.Waves />
          </S.WrapperImg>
        </S.Row>

        <S.ImgPurpleBalloon
          src="/imgs/purple-balloon.svg"
          alt="Balão Roxo"
          width="100"
          height="100"
        />
      </S.Container>

      <C.IF condition={!!children}>
        <S.Footer>{children}</S.Footer>
      </C.IF>
    </>
  )
}

export default Banner
