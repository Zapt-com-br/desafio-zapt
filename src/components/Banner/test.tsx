import { screen } from '@testing-library/react'
import Banner from '.'
import { renderWithTheme } from 'utils/tests/helpers'

describe('<Banner />', () => {
  it('should render the banner', () => {
    const { container } = renderWithTheme(
      <Banner
        title="Bem-vindo Marcio,"
        description="Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar."
        question="O que você prefere?"
        img="/imgs/persons.jpg"
        btn={[
          { textBtn: 'Criar um grupo', route: '/boas-vindas' },
          { textBtn: 'Entrar em um grupo', route: '/boas-vindas' }
        ]}
      />
    )

    const banner = screen.getByAltText('Banner')

    expect(banner).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
