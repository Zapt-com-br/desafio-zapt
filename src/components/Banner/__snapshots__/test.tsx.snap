// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the banner 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c1 {
  background-color: #FAFAFA;
  min-height: 31.3rem;
  position: relative;
}

.c2 {
  display: none;
  position: absolute;
  left: -2.5rem;
  top: -2rem;
  opacity: 0;
  -webkit-animation: hRcgSd 0.8s ease forwards;
  animation: hRcgSd 0.8s ease forwards;
}

.c13 {
  display: none;
  position: absolute;
  right: -4.5rem;
  bottom: -3rem;
  opacity: 0;
  -webkit-animation: dcCKBc 0.8s ease forwards;
  animation: dcCKBc 0.8s ease forwards;
}

.c3 {
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c5 {
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem;
}

.c6 {
  opacity: 0;
  -webkit-animation: FTyau 0.8s ease forwards 0.2s;
  animation: FTyau 0.8s ease forwards 0.2s;
  font-size: 2.8rem;
  font-weight: 500;
}

.c7 {
  opacity: 0;
  -webkit-animation: dcCKBc 0.8s ease forwards 0.2s;
  animation: dcCKBc 0.8s ease forwards 0.2s;
  font-size: 1.6rem;
  font-weight: 400;
  color: #7A7A7A;
  line-height: 1.5;
  padding: 1.2rem 0;
}

.c8 {
  opacity: 0;
  -webkit-animation: kSVEJc 0.8s ease forwards 0.4s;
  animation: kSVEJc 0.8s ease forwards 0.4s;
  font-size: 1.6rem;
  font-weight: 500;
  padding: 1rem 0 1.5rem 0;
}

.c9 {
  gap: 1.6rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c9 .btn0 {
  opacity: 0;
  -webkit-animation: dcCKBc 0.8s ease forwards 0.7s;
  animation: dcCKBc 0.8s ease forwards 0.7s;
}

.c9 .btn1 {
  opacity: 0;
  -webkit-animation: dcCKBc 0.8s ease forwards 0.9s;
  animation: dcCKBc 0.8s ease forwards 0.9s;
}

.c11 {
  position: relative;
  width: 100%;
  min-width: initial;
  max-width: initial;
  aspect-ratio: 1/0.8;
  opacity: 0;
  -webkit-animation: ffSPMP 2s ease forwards;
  animation: ffSPMP 2s ease forwards;
}

.c11 > div {
  position: initial !important;
}

.c11 img {
  object-fit: cover;
}

.c12 {
  display: none;
  background-image: url(/imgs/waves.svg);
  background-repeat: no-repeat;
  background-position: -0.2rem 0;
  background-size: contain;
  position: absolute;
  left: 0;
  bottom: 0;
}

.c10 {
  cursor: pointer;
  border-radius: 0.4rem;
  font-weight: 400;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  border: 0.1rem solid transparent;
  text-shadow: 0 0 0.3rem rgb(0 0 0 / 60%);
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  background-color: #334BC8;
  height: -webkit-min-content;
  height: -moz-min-content;
  height: min-content;
  font-size: 1.4rem;
  padding: 1.6rem 3.2rem;
}

.c10:focus {
  box-shadow: 0 0 0 0.2rem #5872ff,0 0.1rem 1.5rem #3e396b80;
}

.c10:hover {
  background-color: #5872ff;
}

.c10:active {
  box-shadow: none;
}

.c10:disabled,
.c10:disabled:hover {
  opacity: 0.5;
  cursor: not-allowed;
  background-color: #a1a4b0;
}

.c10 span {
  color: #fff;
}

.c10 svg + span {
  margin-left: 1rem;
}

.c10 svg {
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  color: #fff;
}

@media screen and (min-width:800px) {
  .c2 {
    display: block;
  }
}

@media screen and (min-width:800px) {
  .c13 {
    display: block;
  }
}

@media screen and (min-width:800px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:600px) {
  .c5 {
    padding: 3rem 5rem;
  }
}

@media screen and (min-width:998px) {
  .c5 {
    padding: 6rem 10rem;
  }
}

@media screen and (min-width:800px) {
  .c11 {
    min-width: 43.3rem;
    max-width: 43.3rem;
  }
}

@media screen and (min-width:800px) {
  .c12 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    height: 100%;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <img
      alt="Balão Rosa"
      class="c2"
      height="62"
      src="/imgs/pink-balloon.svg"
      width="62"
    />
    <div
      class="c0 c3"
    >
      <div
        class="c4 c5"
      >
        <div
          class="c4 c6"
        >
          Bem-vindo Marcio,
        </div>
        <div
          class="c4 c7"
        >
          Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar.
        </div>
        <div
          class="c4 c8"
        >
          O que você prefere?
        </div>
        <div
          class="c0 c9"
        >
          <button
            aria-label="Criar um grupo"
            class="c10 btn0"
            color="blue"
          >
            <span>
              Criar um grupo
            </span>
          </button>
          <button
            aria-label="Entrar em um grupo"
            class="c10 btn1"
            color="blue"
          >
            <span>
              Entrar em um grupo
            </span>
          </button>
        </div>
      </div>
      <div
        class="c0 c11"
      >
        <img
          alt="Banner"
          height="313"
          src="/imgs/persons.jpg"
          width="433"
        />
        <svg
          class="c12"
        />
      </div>
    </div>
    <img
      alt="Balão Roxo"
      class="c13"
      height="100"
      src="/imgs/purple-balloon.svg"
      width="100"
    />
  </div>
</div>
`;
